tests/createProjectDriver.py

```diff
@@ -0,0 +1,136 @@
+    def download(self):
+        self.driver.find_element_by_link_text("(Download a sample roster files)").click()
+        self.driver.implicitly_wait(10)
+    
+    def setRoster(self, studentFile):
+
+        self.driver.find_element_by_id("student_file").send_keys(studentFile)
+    
+    def setRubrics(self, jsonFile):
+        self.driver.find_element_by_id("json_file").send_keys(jsonFile)
+    
+    def driver_createProject(self, username, password, projectname, projectpassword, 
+        studentFile = "C:/Users/Wangj/Downloads/sample_roster.xlsx", jsonFile = "C:/Users/Wangj/Downloads/teamwork_scale3.json"):
+        
+        logIn.Driver_Login(self,username, password) #login first
+        self.driver.implicitly_wait(5)
+        self.driver.execute_script("arguments[0].click()",self.driver.find_element_by_link_text("Create New Project"))        
+        self.driver.implicitly_wait(5)
+
+        createProject.setProjectName(self, projectname, projectpassword)
+        
+        #createProject.download(self) #the xlsx file should be downloaded previously
+        #time.sleep(2) #this step is necessary to download the file      
+        
+        createProject.setRoster(self, studentFile)
+
+        createProject.setRubrics(self, jsonFile) 
+        
+        #This is for submission
+        self.driver.find_element_by_css_selector(".w3-button").click()
+
+    
+    
+    def createProject_attempt(self, username, password, projectname, projectpassword, studentFile, jsonFile): # if run the 2nd time, the control flow would go to duplicate username
+
+        self.driver_createProject(username, password, projectname, projectpassword, studentFile, jsonFile)  
+
+        urlCurrent = self.driver.current_url
+        
+        try:
+            if self.driver.find_element_by_xpath("/html/body/div[3]/div[2]/div/div/div/form/div[1]/p").text != None:
+                alertInfo = self.driver.find_element_by_xpath("/html/body/div[3]/div[2]/div/div/div/form/div[1]/p").text
+        except:
+            alertInfo = "no error"
+
+        return (urlCurrent, alertInfo)
+        
+        
+    def getProjectNameAndDescriptionAlert(self):
+        alert1 = self.driver.find_element_by_xpath("/html/body/div[3]/div[2]/div/div/div/form/div[1]/p").text
+        alert2 = self.driver.find_element_by_xpath("/html/body/div[3]/div[2]/div/div/div/form/div[2]/p").text
+        self.driver.implicitly_wait(5)
+        return (alert1, alert2)
+        
+    def getInvalidFileAlert(self):
+        self.driver.implicitly_wait(5)
+        alert1 = self.driver.find_element_by_xpath("/html/body/div[3]/div[2]/div/div/div/form/div[3]/p").text
+        alert2 = self.driver.find_element_by_xpath("/html/body/div[3]/div[2]/div/div/div/form/div[4]/p").text
+        self.driver.implicitly_wait(5)
+        return (alert1, alert2)
+    
+    
+    
+    def testRubricFileDriver(self, username, password):
+    
+        logIn.Driver_Login(self, username, password) #login first        
```

Comment:
all method names should be lower case, with underscores (e.g., set_project_name)

Use the refactoring tools in your IDE to do this automatically.

Reply:
All methods now follow this naming convention